core/src/main/java/tech/pegasys/web3signer/core/Eth2Runner.java

```diff
@@ -167,8 +168,16 @@ private ArtifactSignerProvider loadSigners(
             "yaml",
             new YamlSignerParser(List.of(artifactSignerFactory))));
 
-    if (config.getAzureKeyVaultParameters() != null) {
+    if (config.getAzureKeyVaultParameters().isAzureKeyVaultEnabled()) {
       final AzureKeyVaultParameters params = config.getAzureKeyVaultParameters();
+      if (!ObjectUtils.allNotNull(
```

Comment:
Can you add a unit test for this validation check

Reply:
:( At the moment, there are NO unit tests around the Runners - so, can, but it'll be hard...